src/Netflix/Netflix.ts

```diff
@@ -15,33 +27,27 @@ const sendMessage = (payload: Command) =>
     JSON.stringify({ proofKey: TO_PAGE_PROOF_KEY, payload } as Message<Command>)
   );
 
-/**
- * Receive Message
- */
-(() => {
-  window.addEventListener("message", ev => receiveMessage(ev.data));
-  const receiveMessage = (messageStr: string) => {
-    try {
-      const message = JSON.parse(messageStr) as Message<Command>;
-      if (message.proofKey === FROM_PAGE_PROOF_KEY) {
-        const command = message.payload;
-        switch (command.key) {
-          case "changeText":
-            return handleChangeTextAnswer(command);
-          case "changeAudio":
-            return handleChangeAudioAnswer(command);
-        }
-      }
-    } catch {
-      // Silent catch
-      // This try catch is used only to catch JSON.parse failure
-      // Which is impossible to happen within Netflix Plugin scope
-    }
-  };
-})();
-
 // Utilities
 
+const parseJsonOrNull = <T>(maybeJSONString: string): Message<T> | null => {
+  try {
+    return JSON.parse(maybeJSONString);
+  } catch {
+    return null;
+  }
+};
+
+const consumeMessageStringAsCommand = (
```

Comment:
I have a counter-argument for this.

In this plugin's context, a **valid inter-context command message** is defined as:
- a JSON
- has a `proofKey` field with a certain value determined at the start of the javascript runtime
- has a `payload` field with type Command

`consumeMessageStringAsCommand` is used to consume string as a command message and is used in the "message" event on windows. There's a chance Netflix have inter-frame message-passing mechanisms somewhere in their app and those message, while probably are also JSON, can be in other format too because JSON is not in the spec.

So my argument is if a string coming from `window.addEventListener("message", ...)` is not a valid Message<Command> then the plugin should not be concerned with it. Because the plugin should not be concerned with it, any error that comes from it should not be handled, because it is irrelevant. Therefore, in this case, it is better for `consumeMessageStringAsCommand` to silently swallow any errors that come from invalid incoming data.

If it's the debug-friendliness you are concerned with, this scenario fits more into unit-tests rather than runtime-debugging, staying close to "type error should be resolved on compile-time" principle.

That's my two cents, but the final decision will stay in your hands.

What do you think?

Reply:
Actually, I understand what you mean better now. This is good and fine. You can leave it as-is.